pkg/util/vmopv1/vm.go

```diff
@@ -213,3 +216,123 @@ func IsClasslessVM(vm vmopv1.VirtualMachine) bool {
 func IsImagelessVM(vm vmopv1.VirtualMachine) bool {
 	return vm.Spec.Image == nil && vm.Spec.ImageName == ""
 }
+
+func SyncStorageUsageForNamespace(
+	ctx context.Context,
+	k8sClient client.Client,
+	namespace, storageClass string) (bool, <-chan error) {
+
+	var (
+		chanErr = make(chan error, 1)
+		objKey  = client.ObjectKey{
+			Namespace: namespace,
+			Name:      spqutil.StoragePolicyUsageName(storageClass),
+		}
+	)
+
+	syncFn := func() {
+		defer close(chanErr)
+
+		ctx := context.Background()
+
+		// Make sure the StoragePolicyUsage document exists.
+		var obj spqv1.StoragePolicyUsage
+		if err := k8sClient.Get(ctx, objKey, &obj); err != nil {
+
+			// Even if the SPU is not found, log the error to indicate that
+			// the function was enqueued but could not proceed.
+			chanErr <- fmt.Errorf(
+				"failed to get StoragePolicyUsage %s: %w", objKey, err)
+			return
+		}
+
+		var list vmopv1.VirtualMachineList
+		if err := k8sClient.List(
+			ctx,
+			&list,
+			client.InNamespace(namespace)); err != nil {
+
+			chanErr <- fmt.Errorf(
+				"failed to list VMs in namespace %s: %w", namespace, err)
+			return
+		}
+
+		var (
+			totalUsed     resource.Quantity
+			totalReserved resource.Quantity
+		)
+
+		for i := range list.Items {
+			vm := list.Items[i]
+
+			if !vm.DeletionTimestamp.IsZero() || vm.Status.Storage == nil {
+				// Ignore VMs being deleted or that are not reporting storage
+				// status.
+				continue
+			}
+
+			// Account for storage used by this VM.
+			if vm.Status.Storage.Unshared != nil {
+				totalUsed.Add(*vm.Status.Storage.Unshared)
+			}
+
+			// Account for storage marked as reserved by this VM.
+			if vm.Status.Storage.Committed != nil {
+				totalReserved.Add(*vm.Status.Storage.Committed)
+			}
+			if vm.Status.Storage.Uncommitted != nil {
+				totalReserved.Add(*vm.Status.Storage.Uncommitted)
+			}
+
+			// Subtract the PVC usage/limit from the total usage/reserved values
+			// as PVCs are counted separately.
+			for j := range vm.Status.Volumes {
```

Comment:
@SandeepPissay / @xing-yang, just FYI -- 

We are subtracting space used/reserved by PVCs from the VM's total reserved space since PVCs are reported separately. This is what you would expect, correct?

Reply:
Yes.  Sandeep's explanation is right.

Reply:
Ack. I asked Deepak the same internally and did not get a clear answer. I'll update the logic to only mark things reserved if the VM is not yet fully created.

Reply:
@abaruni Based on @SandeepPissay's and @xing-yang's feedback, I will need to rely on the work you are doing for the OVF parsing to get the capacity for VMs not yet fully provisioned. I wonder what the most efficient way to do this is.